### ISSUE TITLE
ULTRA anoying quantum GUI

### ISSUE THREAD

**Issue by NS-Clone** [Feodor2/Mypal68#227]
тут я буду ругaтсо и перечислять мeрзoсти которые нaccточeртeли в этом вашем квантуме

вытягивание вкладок в отдельное окно (tabs detach)
browser.tabs.allowTabDetach;false - не помогает

**Comment by NS-Clone** [comment]
browser.tabs.insertAfterCurrent;true - SUGXXXЪ!!!!11 (not works as expected)
![2023-10-06_030132](https://github.com/Feodor2/Mypal68/assets/139008911/8d45415e-37ec-4dfc-a8c1-9cf834a310f0)

open tabs like whitе mаn аgain
with https://addons.mozilla.org/en-US/firefox/addon/open-tabs-next-to-current/
https://github.com/sblask/webextension-open-tabs-next-to-current/releases
![2023-10-06_030122](https://github.com/Feodor2/Mypal68/assets/139008911/c974883a-c49f-4216-bc9c-a381047f9eca)

**Comment by zanud** [comment]
> пока вкладки правее текущей не будут полностсью загружены? или же с того же сайта
> а обычно новые вкладки открываются именно серией

Согласно моему опыту, логика там такая: новые вкладки открываются одна за одной до тех пор, пока ты остаёшься на вкладке **Current**. Если с **Current** переключиться на любую другую вкладку, а потом вернуться на Current, то начнётся новая серия открывания после **Current**.

> Open tabs like whitе mаn (аgain)
> ![2023-10-06_030122](https://user-images.githubusercontent.com/139008911/273071199-c974883a-c49f-4216-bc9c-a381047f9eca.png)

Мне бы не хотелось, чтобы так было. Лучше уж пусть открываются не справа, а слева от текущей.

А по хорошему, браузеру надо для каждой вкладки помнить, из какой вкладки она была открыта, и при открывании новой вкладки смотреть, есть ли справа вкладки, открытые из текущей. Если есть, то новую добавлять в конец этой непрерывной цепочки.
Но это, скорее всего, потребует серьёзных переделок внутри браузера, так что вряд ли будет реализовано.

**Comment by NS-Clone** [comment]
> Если есть, то новую добавлять в конец этой непрерывной цепочки.

это мерзская мерзоссть
ты жмешь открыть вкладку
а визуального откллика ты не видишь
пушо она вылазит вне поля видимости

мудачье придумавшее такое надо кacтрировать
по крайней мере по дефолту оно должно быть выключено

**Comment by zanud** [comment]
> Это ЕСТЕСТВЕННОЕ и ОЖИДАЕМОЕ поведение
> в упор непонятно КТО? додумалсо все делоть через Ж
> а потом еще и закрывать вкладки с переходом вправо и вправо их же открывать....

Я почти всегда так делаю: иду по страничке сверху вниз и открываю ссылки, которые меня заинтересовали, а потом перехожу на вкладку справа от текущей (на ту, которую первой открывал), читаю её и закрываю, после закрытия автоматом оказываюсь на вкладке, которую открывал второй, читаю и закрываю, после чего автоматом оказываюсь на третьей вкладке и т.д. В результате я просматриваю вкладки именно в том порядке, в котором их открывал, а не в обратном.

> это мерзская мерзоссть
> ты жмешь открыть вкладку
> а визуального откллика ты не видишь
> пушо она вылазит вне поля видимости

Если открывается вне поля видимости, то подмаргивает стрелочка, открывающая список вкладок. Плохо то, что подмаргивает она не сразу при создании вкладки, а когда пошло получение информации от сервера.

И существуют дополнения, показывающие количество вкладок.

А в твоём варианте из поля видимости уйдут первые открытые вкладки, и хрен потом найдёшь ту, которая была открыта первой. А читать вкладки в обратном порядке - _мерзская мерзоссть_.

**Comment by Feodor2** [comment]
Про вкладки тоже сколько помню всегда так было, и это правильно имхо, когда там штук 5 открываеш я хочу чтобы они сдвигались.

Постраничное листание вкладок по двойному клику - никогда с этим не было проблем работает как положено, а в ноутбуки при условии что двойное нажатие только на кнопке, в новых моделях начали тулить двойное нажатие на квдратике, это да я выключаю.

Поэтому я тут ничего делать не стану, но проверю browser.tabs.allowTabDetach

**Comment by NS-Clone** [comment]
> Про вкладки тоже сколько помню всегда так было,

вощето так никогда небыло
это УГ добавили в квантумах

**Comment by zanud** [comment]
> это УГ добавили в квантумах

У меня основной браузер - SeaMonkey, где никаким квантумом и не пахнет, но в нём с открыванием вкладок всё точно так же.

**Comment by NS-Clone** [comment]
> > это УГ добавили в квантумах
> 
> У меня основной браузер - SeaMonkey, где никаким квантумом и не пахнет, но в нём с открыванием вкладок всё точно так же, как в MyPal.

вотт так?
![](https://user-images.githubusercontent.com/139008911/273071066-8d45415e-37ec-4dfc-a8c1-9cf834a310f0.png)

**Comment by zanud** [comment]
Да, именно так. До тех пор, пока остаюсь на **Current**. Если перейду на любую другую вкладку, потом вернусь и продолжу открывать ссылки, то новые будут вставляться между **Current** и **Tab 1**.

**Comment by zanud** [comment]
О, кстати: если в настройках поставить галочку "Переключаться на открываемую вкладку", то порядок вкладок будет именно таким, как ты хочешь - **Current**, **Tab 4**, **Tab 3**, **Tab 2**, ... Только придётся с каждой новооткрытой вкладки возвращаться на **Current**.

**Comment by NS-Clone** [comment]
> У меня основной браузер - SeaMonkey,

МДО...
значит или у меня всегда был tabmixplus
или раньше browser.tabs.insertAfterCurrent работал по другому

а по дефолту до
поведение именно такое
проверил на firefox 5, firefox 31 ESR, firefox 38 ESR, firefox 52 ESR

но я мог принять это гaфнo за открытие вкладок в конце списка вообще
и включить browser.tabs.insertAfterCurrent и 10 лет больше это не видеть

**Comment by NS-Clone** [comment]
FCKN autohide downloads button 🔥 
![185331_](https://github.com/Feodor2/Mypal68/assets/139008911/aac18086-8b1c-4a9b-a595-c12292005af7)
аффтар этой мурни должен DIЕ IN PАIN!

**Comment by NS-Clone** [comment]
наглядный пример почему сдвоенные кнопки ненужны

https://github.com/user-attachments/assets/5259b82a-a75b-4682-9b8c-a104f0b7cf97

**Comment by NS-Clone** [comment]
примерживание нового browser.tabs.allowTabDetach;false все еще сильно реквестируетсо
ибо вкладки выдергивается из браузера просто через раз а иногда и каждый раз
а адон по их втягиванию работает просто чудовищно топорно
и иногда эти вкладки втягивает как самую 1-ю и переводит на нее фокус с 1150-й...

**Comment by NS-Clone** [comment]
все эти open right close left адоны тоже работают через Ж
в итоге направление открытия закрытия все время непредсказуемое
при старте браузера пока пару раз не закроешь
оно вообще не закрывает в нужную сторону
